### PR TITLE
add lido gateway / fetch l1gateway if not included

### PR DESCRIPTION
**Pull request by DZGoldman**

### CODE REVIEW

File: src/lib/token_list_gen.ts [hunk 1]
```diff
@@ -285,7 +289,31 @@ export const generateTokenList = async (
     if (uri) logoUris[token.l1TokenAddr] = uri;
   }
 
-  let arbifiedTokenList: ArbTokenInfo[] = tokens
+  const getL1GatewayAddress = async (l2GatewayAddress: string) => {
+    let l2Gateway: string;
+    if (isNova) {
+      if (
+        (l2Gateway = l2ToL1GatewayAddressesNova[l2GatewayAddress.toLowerCase()])
```
Comment (fionnachan) [inline src/lib/token_list_gen.ts:296]: is this line supposed to be making an assignment to `l2Gateway` or a conditional statement?

Comment (DZGoldman) [reply src/lib/token_list_gen.ts:296]: yeah - assignment + pass truthy conditional if in mapping, assign undefined so falsy if not in mapping

File: src/lib/token_list_gen.ts [hunk 2]
```diff
@@ -285,7 +289,31 @@ export const generateTokenList = async (
...
+      ) {
+        return l2Gateway;
+      }
+    } else {
+      if ((l2Gateway = l2ToL1GatewayAddresses[l2GatewayAddress.toLowerCase()]))
+        return l2Gateway;
+    }
```
Comment (fionnachan) [inline src/lib/token_list_gen.ts:303]:
```suggestion
    let l2Gateway: string = isNova
      ? l2ToL1GatewayAddressesNova[l2GatewayAddress.toLowerCase()]
      : l2ToL1GatewayAddresses[l2GatewayAddress.toLowerCase()];
    if (l2Gateway) {
      return l2Gateway;
    }
```
Personally I'd prefer this for better readability 🧐
or add a comment for future readers
but I'll defer it to others

**Review by gzeoneth — APPROVED** [review]
LGTM, @fionnachan 's refactor also looks better but maybe in another PR since there are some urgency here.

**Pull request merged**